packages/rspack/src/utils/read-rspack-options.ts

```diff
@@ -5,53 +5,66 @@ import { readNxJsonFromDisk } from 'nx/src/devkit-internals';
 
 /**
  * Reads the Rspack options from a give Rspack configuration. The configuration can be:
- * 1. A standard config object
- * 2. A standard function that returns a config object
- * 3. A Nx-specific composable function that takes Nx context, rspack config, and returns the config object.
+ * 1. A single standard config object
+ * 2. A standard function that returns a config object (standard Rspack)
+ * 3. An array of standard config objects (multi-configuration mode)
+ * 4. A Nx-specific composable function that takes Nx context, rspack config, and returns the config object.
  *
  * @param rspackConfig
  */
 export async function readRspackOptions(
   rspackConfig: unknown
-): Promise<Configuration> {
-  let config: Configuration;
-  if (isNxRspackComposablePlugin(rspackConfig)) {
-    config = await rspackConfig(
-      {},
-      {
-        // These values are only used during build-time, so passing stubs here just to read out
-        // the returned config object.
-        options: {
-          root: workspaceRoot,
-          projectRoot: '',
-          sourceRoot: '',
-          outputFileName: '',
-          assets: [],
-          main: '',
-          tsConfig: '',
-          outputPath: '',
-          rspackConfig: '',
-          useTsconfigPaths: undefined,
-        },
-        context: {
-          root: workspaceRoot,
-          cwd: undefined,
-          isVerbose: false,
-          nxJsonConfiguration: readNxJsonFromDisk(workspaceRoot),
-          projectGraph: null,
-          projectsConfigurations: null,
+): Promise<Configuration[]> {
+  const configs: Configuration[] = [];
+
+  const resolveConfig = async (config: Configuration) => {
+    if (isNxRspackComposablePlugin(rspackConfig)) {
+      config = await rspackConfig(
+        {},
+        {
+          // These values are only used during build-time, so passing stubs here just to read out
+          // the returned config object.
+          options: {
+            root: workspaceRoot,
+            projectRoot: '',
+            sourceRoot: '',
+            outputFileName: '',
+            assets: [],
+            main: '',
+            tsConfig: '',
+            outputPath: '',
+            rspackConfig: '',
+            useTsconfigPaths: undefined,
+          },
+          context: {
+            root: workspaceRoot,
+            cwd: undefined,
+            isVerbose: false,
+            nxJsonConfiguration: readNxJsonFromDisk(workspaceRoot),
+            projectGraph: null,
+            projectsConfigurations: null,
+          },
+        }
+      );
+    } else if (typeof rspackConfig === 'function') {
```

Comment:
Is it valid to export a function that returns multiple config objects? Or can you export an array that contains functions? Or both?

It's not clear from the webpack/rspack docs which is a valid mix of array and function, can you double-check this case?

Reply:
Each element in the array can be either a function or a config object.
Each function returns a config object.

Reply:
So webpack/rspack do not support a function that returns an array?

Reply:
It does, I have updated the PR to handle these cases as well.

Reply:
> So webpack/rspack do not support a function that returns an array?
You can: https://github.com/Coly010/ng-rspack-build/blob/main/packages/build/src/lib/config/create-config.ts#L12
Usage: https://github.com/Coly010/ng-rspack-build/blob/main/apps/rspack/ssr/css/rspack.config.js#L3

Reply:
This is supported with the PR.

The error you will get is if your `module.exports` is a function that returns an array which has 1 or more functions.